src/Filter/ChoiceFilter.php

```diff
@@ -35,29 +35,57 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
                 return;
             }
 
+            $isNullSelected = \in_array(null, $data['value'], true);
+            $data['value'] = array_filter($data['value'], static function ($value) {
+                return null !== $value;
+            });
+
             // Have to pass IN array value as parameter. See: http://www.doctrine-project.org/jira/browse/DDC-3759
             $completeField = sprintf('%s.%s', $alias, $field);
             $parameterName = $this->getNewParameterName($queryBuilder);
             if (ChoiceType::TYPE_NOT_CONTAINS === $data['type']) {
-                $this->applyWhere($queryBuilder, $queryBuilder->expr()->notIn($completeField, ':'.$parameterName));
+                if (0 === \count($data['value'])) {
+                    $this->applyWhere($queryBuilder, $queryBuilder->expr()->isNotNull($completeField));
+                } else {
+                    $this->applyWhere($queryBuilder, $queryBuilder->expr()->andX(
+                        $queryBuilder->expr()->notIn($completeField, ':'.$parameterName),
+                        $queryBuilder->expr()->isNotNull($completeField)
+                    ));
+                    $queryBuilder->setParameter($parameterName, $data['value']);
+                }
             } else {
-                $this->applyWhere($queryBuilder, $queryBuilder->expr()->in($completeField, ':'.$parameterName));
+                if ($isNullSelected) {
+                    $this->applyWhere($queryBuilder, $queryBuilder->expr()->orX(
+                        $queryBuilder->expr()->in($completeField, ':'.$parameterName),
+                        $queryBuilder->expr()->isNull($completeField)
+                    ));
```

Comment:
Empty data array will result in an incorrect query here.
```sql
SELECT * FROM ... WHERE field IN () OR field IS NULL
```

Reply:
Empty data array will no result in an incorrect query.
Empty data are always replaced by `[null]`.

And we get the following query
```
SELECT * FROM ... WHERE field IN (NULL) OR field IS NULL
```

Reply:
Yes. You're right. I'm confusing something.